gslib/utils/copy_helper.py

```diff
@@ -1612,7 +1617,12 @@ def _SetContentTypeFromFile(src_url, dst_obj_metadata):
               'Encountered OSError running "file -b --mime %s"\n%s' %
               (real_file_path, e))
       else:
-        content_type = mimetypes.guess_type(real_file_path)[0]
+        for extension, extension_content_type in COMMON_EXTENSION_RULES.items():
```

Comment:
This loop feels like it should be a set or dict look-up.
Can you do something like:
```
split_file_path = real_file_path.split('.')
if len(split_file_path) > 1:
  extension = split_file_path[-1]
  if extension in COMMON_EXTENSION_RULES:
    ...
```

That might break if there are double-extensions like `.txt.tar`, but it should work with all the currently supported rules.

Reply:
yeah, adding double-extension rules was exactly what I was worried about. Made the change though.